src/graph/validator/FindPathValidator.cpp

```diff
@@ -46,6 +46,7 @@ Status FindPathValidator::validateWhere(WhereClause* where) {
   }
   where->setFilter(ExpressionUtils::rewriteLabelAttr2EdgeProp(expr));
   auto filter = where->filter();
+  NG_RETURN_IF_ERROR(checkExprDepth(filter));
 
   auto typeStatus = deduceExprType(filter);
```

Comment:
It seems that you only need to do this `checkExprDepth()` inside `deduceExprType()`.  So you don't have to add this check everywhere.

Reply:
done